File: src/AddressSync/Forms/FrmFlightImport.vb
Public Class FrmFlightImport

    Public Sub New()
        InitializeComponent()
        KeyPreview = True

    End Sub

    ' Wenn Enter bzw. ESC geklickt wird
    Private Sub FrmSettings_KeyDown(sender As Object, e As KeyEventArgs) Handles MyBase.KeyDown
        If e.KeyData = Keys.Escape Then
            DialogResult = DialogResult.OK
        ElseIf e.KeyData = Keys.Enter Then
            DialogResult = DialogResult.Cancel
        End If
    End Sub
End Class

File: src/AddressSync/Forms/FrmFlightImport.cs
using System.Windows.Forms;

namespace FlsGliderSync
{
    public partial class FrmFlightImport
    {
        public FrmFlightImport()
        {
            InitializeComponent();
            KeyPreview = true;
        }

        // Wenn Enter bzw. ESC geklickt wird
        private void FrmSettings_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyData == Keys.Escape)
            {
                DialogResult = DialogResult.OK;
            }
            else if (e.KeyData == Keys.Enter)
            {
                DialogResult = DialogResult.Cancel;
            }
        }
    }
}